src/coreComponents/constitutive/fluid/PVTFunctions/FenghourCO2ViscosityFunction.cpp

```diff
@@ -16,155 +16,149 @@
  * @file FenghourCO2ViscosityFunction.cpp
  */
 
-
 #include "constitutive/fluid/PVTFunctions/FenghourCO2ViscosityFunction.hpp"
+
 #include "constitutive/fluid/PVTFunctions/SpanWagnerCO2DensityFunction.hpp"
+#include "managers/Functions/FunctionManager.hpp"
+#include "managers/GeosxState.hpp"
 
 namespace geosx
 {
 
-namespace PVTProps
+namespace constitutive
 {
 
-FenghourCO2ViscosityFunction::FenghourCO2ViscosityFunction( string_array const & inputPara,
-                                                            string_array const & componentNames,
-                                                            real64_array const & componentMolarWeight ):
-  PVTFunction( inputPara[1], componentNames, componentMolarWeight )
+namespace PVTProps
 {
 
+FenghourCO2Viscosity::FenghourCO2Viscosity( array1d< string > const & inputPara,
+                                            array1d< string > const & componentNames,
+                                            array1d< real64 > const & componentMolarWeight ):
+  PVTFunctionBase( inputPara[1],
+                   componentNames,
+                   componentMolarWeight )
+{
   makeTable( inputPara );
-
 }
 
-void FenghourCO2ViscosityFunction::makeTable( string_array const & inputPara )
+void FenghourCO2Viscosity::makeTable( array1d< string > const & inputPara )
 {
+  array1d< array1d< real64 > > coordinates;
+  coordinates.resize( 2 );
 
-  real64_array pressures;
-  real64_array temperatures;
-
-  real64 PStart, PEnd, dP;
-  real64 TStart, TEnd, dT;
-  real64 P, T;
-
-  dT = -1.0;
-  dP = -1.0;
-  TStart = -1.0;
-  TEnd = -1.0;
-  PStart = -1.0;
-  PEnd = -1.0;
+  real64 TStart = -1.0;
+  real64 TEnd = -1.0;
+  real64 dT = -1.0;
+  real64 PStart = -1.0;
+  real64 PEnd = -1.0;
+  real64 dP = -1.0;
```

Comment:
Here I left the code as it was, because when I do the modifications you suggest I get `error: 'PStart' may be used u\
ninitialized in this function [-Werror=maybe-uninitialized]`